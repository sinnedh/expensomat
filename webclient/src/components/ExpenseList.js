import React from 'react';
import ExpenseListItem from './ExpenseListItem'

export default (props) => (
  <ul>
    {Object.keys(props.expenses).map((id) =>
      <ExpenseListItem
        key={id}
        expense={props.expenses[id]}
        onClickDelete={props.onClickDelete}
        />
      )
    }
  </ul>
)
