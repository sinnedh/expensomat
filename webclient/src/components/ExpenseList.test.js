import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router'
import ExpenseList from './ExpenseList';

const expenses = [
  { description: "Pasta", amount: 800, paid_for: [], paid_by: [], paid_at: "2017-11-11 15:28"},
  { description: "Pizza", amount: 1200, paid_for: [], paid_by: [], paid_at: "2016-09-14 12:18"},
]

describe('ExpenseListItem', () => {
  it('renders when no expenses are given', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ExpenseList expenses={[]} />, div);
  });

  it('matches snapshop when canEditExpenses=true', () => {
    const context = { user: {canEditExpenses: true} }
    const wrapper = shallow(<ExpenseList expenses={expenses} />, {context})
    expect(wrapper).toMatchSnapshot()
  });

  it('matches snapshop when canEditExpenses=false', () => {
    const context = { user: {canEditExpenses: false} }
    const wrapper = shallow(<ExpenseList expenses={expenses} />, {context})
    expect(wrapper).toMatchSnapshot()
  });
})
