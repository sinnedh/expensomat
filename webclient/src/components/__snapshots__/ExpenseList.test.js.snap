// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseListItem matches snapshop when canEditExpenses=false 1`] = `
<ul>
  <ExpenseListItem
    expense={
      Object {
        "amount": 800,
        "description": "Pasta",
        "paid_at": "2017-11-11 15:28",
        "paid_by": Array [],
        "paid_for": Array [],
      }
    }
    key="0"
  />
  <ExpenseListItem
    expense={
      Object {
        "amount": 1200,
        "description": "Pizza",
        "paid_at": "2016-09-14 12:18",
        "paid_by": Array [],
        "paid_for": Array [],
      }
    }
    key="1"
  />
</ul>
`;

exports[`ExpenseListItem matches snapshop when canEditExpenses=true 1`] = `
<ul>
  <ExpenseListItem
    expense={
      Object {
        "amount": 800,
        "description": "Pasta",
        "paid_at": "2017-11-11 15:28",
        "paid_by": Array [],
        "paid_for": Array [],
      }
    }
    key="0"
  />
  <ExpenseListItem
    expense={
      Object {
        "amount": 1200,
        "description": "Pizza",
        "paid_at": "2016-09-14 12:18",
        "paid_by": Array [],
        "paid_for": Array [],
      }
    }
    key="1"
  />
</ul>
`;
